src/sentry/migrations/0286_backfill_alertrule_organization.py

```diff
@@ -0,0 +1,58 @@
+# Generated by Django 2.2.27 on 2022-04-08 22:24
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def backfill_alertrule_organization(apps, schema_editor):
+    AlertRule = apps.get_model("sentry", "AlertRule")
+    Organization = apps.get_model("sentry", "Organization")
+
+    for alert_rule in RangeQuerySetWrapperWithProgressBar(AlertRule.objects_with_snapshots.all()):
+        if not alert_rule.organization_id:
+            continue
+
+        correct_org = (
+            Organization.objects.filter(
+                project__querysubscription__snuba_query__alertrule__id=alert_rule.id
+            )
+            .distinct()
+            .get()
+        )
```

Comment:
It's probably worth catching `Organization.DoesNotExist` here and continuing, just in case there are orphaned alert rules with no subscriptions hanging around.